app/scripts/lib/message-manager.js

```diff
@@ -180,6 +180,17 @@ export default class MessageManager extends EventEmitter {
    * @returns {Promise<object>} Promises the msgParams object with metamaskId removed.
    */
   approveMessage(msgParams) {
+    const msg = this.getMsg(msgParams.metamaskId);
+
+    this.metricsEvent({
+      event: 'Sign Request',
+      category: EVENT.CATEGORIES.TRANSACTIONS,
+      properties: {
+        action: 'Sign Request Approve',
+        type: msg.type,
+      },
+    });
+
```

Comment:
This code needs to be removed, events for approval are already tracked in createRPCMethodTrackingMiddleware